analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AssignmentExpressionSyntaxExtensionsTests.cs

```diff
@@ -27,6 +27,356 @@ namespace SonarAnalyzer.UnitTest.Extensions
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new
+                        {
+                            Arguments = new[]
+                            {
+                                new { Expression = new { Designation = new { Identifier = new { Text = "x" } } } },
+                                new { Expression = new { Designation = new { Identifier = new { Text = "y" } } } },
+                                new { Expression = new { Designation = new { Identifier = new { Text = "z" } } } },
+                            }
+                        },
+                        Right =new
+                        {
+                            Arguments = new[]
+                            {
+                                new { Expression = new { Token = new { Text = "1" } } },
+                                new { Expression = new { Token = new { Text = "2" } } },
+                            }
+                        },
+                    },
+                });
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_MapAssignmentArguments_MisalignedNested1()
+        {
+            var code = "(var a, (var b, var c)) = (1, (2, 3, 4));";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new { Arguments = new { Count = 2 } },
+                        Right = new { Arguments = new { Count = 2 } },
+                    },
+                });
+            mapping[0].Left.ToString().Should().Be("(var a, (var b, var c))");
+            mapping[0].Right.ToString().Should().Be("(1, (2, 3, 4))");
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_MapAssignmentArguments_MisalignedNested2()
+        {
+            var code = "(var a, (var b, var c), var d) = (1, (2, 3, 4));";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new { Arguments = new { Count = 3 } },
+                        Right = new { Arguments = new { Count = 2 } },
+                    },
+                });
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_MapAssignmentArguments_DifferentConventions()
+        {
+            var code = @"int a;
+                         M() => 1;
+                         ((a), (var b, _), object c, double d, _) = (1, (two: 2, (3)), string.Empty, M(), new object());";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
```

Comment:
This method is too hard to read. Let's use new(left, right) constructor. The `Create*` helpers will help a lot too.